src/main/java/net/snowflake/client/core/bind/BindUploader.java

```diff
@@ -166,11 +173,15 @@ private synchronized String synchronizedTimestampFormat(String o) {
    * Create a new BindUploader which will upload to the given stage path Ensure temporary directory
    * for file writing exists
    *
+   * <p>Deprecated now that streaming uploads are used instead of temporary file uploads. BindDir is
+   * not initialized anymore.
+   *
    * @param session the session to use for uploading binds
    * @param stageDir the stage path to upload to
    * @return BindUploader instance
    * @throws BindException if temporary directory could not be created
    */
+  @Deprecated
   public static synchronized BindUploader newInstance(SFSession session, String stageDir)
```

Comment:
Why not modify this method directly instead of deprecating it? Is anyone still using it?

Reply:
No, no one is using it. I could probably just delete all old implementations and tests for the old implementation. I just thought it was common practice to deprecate older functions instead of deleting entirely, but I guess this is all under the hood so it's probably okay to just delete?

Reply:
In my option, `@Deprecated` is used for allowing customers or any developers to keep using these old ones without blocking their code. I guess here we don't have that concern since all old ones have been replaced.

Reply:
Ok, makes sense. Took out all unused code.